telegram_phone_number_checker/main.py

```diff
@@ -12,23 +12,30 @@ def get_names(client, phone_number):
     result = {}
     print(f'Checking: {phone_number=} ...', end="", flush=True)
     try:
+        # Create a contact
         contact = InputPhoneContact(client_id = 0, phone = phone_number, first_name="", last_name="")
+        # Attempt to add the contact from the address book
         contacts = client(functions.contacts.ImportContactsRequest([contact]))
-        username = contacts.to_dict()['users'][0]['username']
-        if not username:
-            result.update({"error": f'ERROR: no username detected'})
-            del_usr = client(functions.contacts.DeleteContactsRequest(id=[username]))
-        else:
-            result.update({"username": username})
-            del_usr = client(functions.contacts.DeleteContactsRequest(id=[username]))
+
+        users = contacts.to_dict()['users']
+        number_of_matches = len(users)
+
+        if number_of_matches == 0:
+            result.update({"error": f'No response, the phone number is not on Telegram or has blocked contact adding.'})
+        elif number_of_matches == 1:
+            user = users[0] 
+            # Attempt to remove the contact from the address book
+            client(functions.contacts.DeleteContactsRequest(id=[user['id']]))
             # getting more information about the user
-            id = del_usr.to_dict()['users'][0]['id']
-            first_name = del_usr.to_dict()['users'][0]['first_name']
-            last_name = del_usr.to_dict()['users'][0]['last_name']
-            result.update({"first_name": first_name, "last_name": last_name, "id": id})
+            result.update({
+                "id": user['id'],
+                "username": user['username'],
```

Comment:
```suggestion
                "username": user.get('username'),
```
Suggesting using `get` to avoid KeyErrors. I have not tested the code so this suggestion is redundant if you've tested that `"username" in user == True`. Maybe we can add `.get(` for first/last names too out of precaution.

Reply:
Great shout, the API should always return `None` values for these, but `get` is better practice.